renderdoc/driver/vulkan/vk_resources.h

```diff
@@ -1071,6 +1071,47 @@ struct MemRefs
 struct ImgRefs;
 struct ImageState;
 
+class VkPendingSubmissionCompleteCallbacks
+{
+public:
+  VkPendingSubmissionCompleteCallbacks() = default;
+  VkPendingSubmissionCompleteCallbacks(VkDevice device, VkEvent event,
+                                       rdcarray<std::function<void()>> callbacks);
+  VkPendingSubmissionCompleteCallbacks(const VkPendingSubmissionCompleteCallbacks &other);
+  VkPendingSubmissionCompleteCallbacks(VkPendingSubmissionCompleteCallbacks &&other);
+  VkPendingSubmissionCompleteCallbacks &operator=(VkPendingSubmissionCompleteCallbacks other);
+
+  ~VkPendingSubmissionCompleteCallbacks();
+
+  operator VkEvent() const
+  {
+    SCOPED_LOCK(m_Lock);
+    return d == NULL ? VK_NULL_HANDLE : d->event;
+  }
```

Comment:
I think rather than having an implicit conversion operator the event should just be a public member for use like `callbacks`.